go/fn/object_test.go

```diff
@@ -10,6 +10,78 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+type GVKTest struct {
+	group   string
+	version string
+	kind    string
+	resultA bool
+	resultB bool
+	resultC bool
+}
+
+func TestGVK(t *testing.T) {
```

Comment:
nit: suggest calling this `TestIsGVK` (it's easier to know exactly which function this is testing if you include the function name in the test name)